text/0000-bundle-groups.md

```diff
@@ -0,0 +1,89 @@
+- Feature Name: groups\_command
+- Start Date: 2017-08-05
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+This is a proposal to deprecate the `with` and `without` settings, replacing them with a `bundle groups` command to list groups and manage whether each group is enabled or disabled for future runs of `bundle install`.
+
+# Motivation
+
+Enabling and disabling groups has been a consistent thorn in the side of Bundler’s UX ever since we first shipped 1.0. We only supported `--without` for more than five years. After more than five years of relentless user requests, we finally caved and agreed to add `group: :foo, optional: true` and the `--with` option.
+
+In Bundler 2.0, the `--with` and `--without` options will (at a minimum) no longer be remembered, and they may be removed entirely. Instead, we currently tell users to instead run `bundle config with` and `bundle config without`, respectively.
+
+In one stroke, adding a `bundle groups` command would replace both flags, both config commands, and open up the possibility of a _not_ remembered `install --only` to satisfy [the requests for `--only`]. It also provides users with an instant readout of what they will actually get when they run `bundle install` without having to do mental gymnastics combining the output of `config`. (Or even worse, give up and try it out, and then get surprised by the results.)
+
+# Guide-level explanation
+
+### Migration
+
+For users coming from Bundler 1, the `groups` command replaces the old options `--with` and `--without`, which allowed you to enable or disable groups on the fly while running `install`. In Bundler 1.99, the `--with` and `--without` options will print a deprecation warning when used, directing users to switch to using the `groups` command instead.
+
+### New `groups` command
+
+The `bundle groups` command allows you to list groups of gems declared in the Gemfile, and find out whether they will be installed or not. It also lets you enable or disable groups for future installs.
+
+Run the command `bundle groups` to list the groups defined in your Gemfile and find out whether they are enabled or disabled. Enabled groups will be installed when you run `bundle install`, and disabled groups will not be installed. Groups will be enabled by default, but they can be disabled by default by setting `optional: true` inside the Gemfile.
```

Comment:
Will there be an option to combine `groups` and `install` in one step? 
As a power user, I might want to change the `groups` and `install` in one step instead of two.
Idea: `bundle groups enable test --install`

Reply:
That's what the existing `--with` and `--without` options for the `install` command do today. It's possible that we could replace both `--with` and `--without` with a single `--only` option, and that would do exactly what you're describing.